api/core/v1alpha1/clustercatalog_types.go

```diff
@@ -68,82 +69,140 @@ type ClusterCatalogList struct {
+	//   - "Deleted", represents whether, or not, the catalog contents have been successfully deleted.
+	//
+	// The current set of reasons are:
+	//   - "UnpackPending", this reason is set on the "Unpack" condition when unpacking the catalog has not started.
+	//   - "Unpacking", this reason is set on the "Unpack" condition when the catalog is being unpacked.
+	//   - "UnpackSuccessful", this reason is set on the "Unpack" condition when unpacking the catalog is successful and the catalog metadata is available to the cluster.
+	//   - "FailedToStore", this reason is set on the "Unpack" condition when an error has been encountered while storing the contents of the catalog.
+	//   - "FailedToDelete", this reason is set on the "Delete" condition when an error has been encountered while deleting the contents of the catalog.
+	//
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
-
-	// resolvedSource contains information about the resolved source
+	// resolvedSource contains information about the resolved source based on the source type.
+	//
+	// Below is an example of a resolved source for an image source:
+	// resolvedSource:
+	//
+	//  image:
+	//    lastPollAttempt: "2024-09-10T12:22:13Z"
+	//    lastUnpacked: "2024-09-10T12:22:13Z"
+	//    ref: quay.io/operatorhubio/catalog:latest
+	//    resolvedRef: quay.io/operatorhubio/catalog@sha256:c7392b4be033da629f9d665fec30f6901de51ce3adebeff0af579f311ee5cf1b
+	//  type: image
 	// +optional
 	ResolvedSource *ResolvedCatalogSource `json:"resolvedSource,omitempty"`
-	// contentURL is a cluster-internal address that on-cluster components
-	// can read the content of a catalog from
+	// contentURL is a cluster-internal URL from which on-cluster components
+	// can read the content of a catalog
 	// +optional
 	ContentURL string `json:"contentURL,omitempty"`
 	// observedGeneration is the most recent generation observed for this ClusterCatalog. It corresponds to the
 	// ClusterCatalog's generation, which is updated on mutation by the API Server.
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
-
 	// LastUnpacked represents the time when the
 	// ClusterCatalog object was last unpacked.
 	// +optional
 	LastUnpacked metav1.Time `json:"lastUnpacked,omitempty"`
 }
 
-// CatalogSource contains the sourcing information for a Catalog
+// CatalogSource is a discriminated union of possible sources for a Catalog.
 type CatalogSource struct {
-	// type defines the kind of Catalog content being sourced.
+	// type is a required reference to the type of source the catalog is sourced from.
+	//
+	// Allowed values are ["image"]
+	//
+	// When this field is set to "image", the ClusterCatalog content will be sourced from an OCI image.
+	// When using an image source, the Image field must be set and must be the only field defined for this type.
```

Comment:
Field reference should be lowercase to match the json tag:
```suggestion
	// When using an image source, the image field must be set and must be the only field defined for this type.
```

Even better might be to include the full field path:
```suggestion
	// When using an image source, the .spec.source.image field must be set and must be the only field defined for this type.
```